### PR TITLE
[Hotfix]use production build profile for the archive binary

### LINKED ISSUE #8080: Coinbase amount incorrect in archive db 1.0.0-fd39808 / finalfinal2
Archive node 1.0.0-fd39808 on Network finalfinal2 is storing coinbase reward as 20000000000, expected 720mina.

```SQL
        SELECT
            bic.block_id,
            sum( CASE WHEN ic.type = 'coinbase' THEN coalesce( ic.fee, 0) ELSE 0 END ) AS coinbase,
            sum( CASE WHEN ic.type = 'fee_transfer_via_coinbase' THEN coalesce( ic.fee, 0) ELSE 0 END) AS feeTransferFromCoinbase,
            max( CASE WHEN ic.type = 'coinbase' THEN ic.receiver_id ELSE NULL END) AS coinbaseReceiverId
        FROM blocks_internal_commands bic 
        INNER JOIN internal_commands ic ON bic.internal_command_id = ic.id
        GROUP BY bic.block_id;
```
"block_id","coinbase","feetransferfromcoinbase","coinbasereceiverid"
3,20000000000,0,7
4,20000000000,0,8
2,20000000000,0,4

### LINKED ISSUE #8082: mina-archive does not work
1) Running Block Producer as docker image **minaprotocol/mina-daemon-baked:1.0.0-fd39808**
2) Running mina-archive as daemon **mina-archive=1.0.0-fd39808**
Error:
```
{
      "sexp": [
        [
          "rpc_error",
          [
            "Bin_io_exn",
            [
              [ "location", "server-side rpc query un-bin-io'ing" ],
              [
                "exn",
                [
                  "Failure",
                  "bin_read_t: version read 260000000 does not match expected version 1"
                ]
              ]
            ]
          ]
        ],
        [ "connection_description", "<created-directly>" ],
        [ "rpc_tag", "Send_archive_diff" ],
        [ "rpc_version", "0" ]
      ]
    }
  ],
  "length": 5,
  "tags": [
    {
      "tag":
        "Could not send archive diff data to archive process after 5 tries. The process may not be running, please check the daemon-argument",
      "sexp": [
        [ "host_and_port", [ "host.docker.internal", "3086" ] ],
        [ "daemon-argument", "--archive-address" ]
      ]
    }
```
In the previous testnet everything worked fine but now it does not.

### PR DESCRIPTION
**Pull request by deepthiskumar**
DUNE_PROFILE is not being set when building the archive.exe causing versioning mismatch between daemon and the archive process.

Fixes #8082
Fixes #8080

### CODE REVIEW

**Review by lk86 — APPROVED** [review]
yeah that works for me

File: Makefile [hunk 1]
```diff
@@ -83,7 +83,7 @@ build: git_hooks reformat-diff libp2p_helper
 
 build_archive: git_hooks reformat-diff
 	$(info Starting Build)
-	ulimit -s 65532 && (ulimit -n 10240 || true) && dune build src/app/archive/archive.exe --profile=$(DUNE_PROFILE)
+	ulimit -s 65532 && (ulimit -n 10240 || true) && dune build src/app/archive/archive.exe --profile=testnet_postake_medium_curves
```
Comment (mrmr1993) [inline Makefile:86]: Can you change the profiles in the CI jobs instead? This will clobber the build artifacts when using this during development.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (deepthiskumar) [reply Makefile]: I think the change in 4d26dac should do

**Review by mrmr1993 — APPROVED** [review]
Looks good to me, thanks!

**Comment by deepthiskumar** [comment]
Tested  on one of our archive nodes (archive1) and blocks (with snarks in them) are getting added. Also verified that coinbase amount in the database is 720

**Review by 0x0I — APPROVED** [review]

**Comment by es92** [comment]
!approved-for-mainnet

**Pull request merged**